File: _assets/scripts/normalizers/materials_normalize_material.lua
return function (data)
    local data_attr = python.as_attrgetter(data)
    if data_attr.__contains__("materials") then
        data["version"] = data["materials"]["version"]
    else
        local output = python.builtins.dict()
        output["materials"] = data
        output["defined_in"] = data["defined_in"]
        return output
    end
end

File: _assets/scripts/normalizers/materials_normalize_material.py
from typing import Any, TypedDict, cast

__all__ = ["materials_normalize_material"]

input_typed_dict = TypedDict("input_typed_dict", version=str, materials=dict[str,Any])

output_typed_dict = TypedDict("output_typed_dict", materials=dict[str,Any], defined_in=list[str])

def materials_normalize_material(data:input_typed_dict|dict[str,Any]) -> output_typed_dict|None:
    if "materials" in data:
        data["version"] = cast(str, data["materials"]["version"])
        del data["materials"]["version"]
    else:
        output:output_typed_dict = {
            "materials": data,
            "defined_in": cast(list[str], data["defined_in"]),
        }
        del output["materials"]["defined_in"]
        return output
